apps/client/src/components/Profile/ProfileSettings.tsx

```diff
@@ -0,0 +1,159 @@
+import { useNavigate } from "@tanstack/react-router";
+import Button from "../Button/Button";
+import { UserProfile } from "@shared/userProfile.ts";
+import FormInput from "../form-components/FormInput";
+import { useForm, FieldValues } from "react-hook-form";
+import styles from "./ProfileSettings.module.css";
+import { Grid } from "@mantine/core";
+import DragAndDrop from "../DragAndDrop/DragAndDrop";
+import FormTextarea from "../form-components/Textarea";
+import { useEffect, useState } from "react";
+import axiosInstance from "../../api/axiosInstance";
+import axios from "axios";
+
+interface ProfileInfoProps {
+  user: UserProfile;
+}
+const ProfileSettings: React.FC<ProfileInfoProps> = ({ user }) => {
+  const navigate = useNavigate();
+  const { control, handleSubmit, setValue } = useForm<FieldValues>({
+    defaultValues: {
+      name: user.name || "",
+      surname: user.surname || "",
+      email: user.email || "",
+      phone: user.phone || "",
+      postcode: user.location?.postCode || "",
+      city: user.location?.city || "",
+      profileText: user.profileText || "",
+      isSeeking: user.isSeeking || false,
+    },
+  });
+  const [isSeeking, setIsSeeking] = useState(user.isSeeking ?? false);
+
+  // Update form when isSeeking changes
+  useEffect(() => {
+    setValue('isSeeking', isSeeking);
+  }, [isSeeking, setValue]);
+  
+  const onSubmit = async (data: FieldValues) => {
+    try {
+      const updateData = {
+        name: data.name,
+        surname: data.surname,
+        email: data.email,
+        phone: data.phone,
+        location: {
+          postCode: data.postcode,
+          city: data.city
+        },
+        profileText: data.profileText,
+        isSeeking: data.isSeeking
+      };
+
+      await axiosInstance.patch(`/users/${user.email}`, updateData);
```

Comment:
Can we move this to userApi.ts? so we follow the same pattern of having all the api calls in the api directory. For the image upload you will need different headers, like this:    'Content-Type': 'multipart/form-data',